roachtest: schemachange/mixed-versions failed

[(roachtest).schemachange/mixed-versions failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=2172497&tab=buildLog) on [master@71566fd4fa0a4efd579a2fb46c3738f92b5a4dbc](https://github.com/cockroachdb/cockroach/commits/71566fd4fa0a4efd579a2fb46c3738f92b5a4dbc):

```
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachtest/test_runner.go:754
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1357
		Wraps: (2) 2 safe details enclosed
		Wraps: (3) output in run_064147.122_n2_workload_run_schemachange
		Wraps: (4) /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod run teamcity-2172497-1597213671-09-n4cpu4:2 -- ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 100 --concurrency 5 {pgurl:1-4} returned
		  | stderr:
		  | eating load: failed to initialize the load generator: dial tcp 10.128.1.2:26257: connect: connection refused
		  | W200812 06:51:46.755753 1 workload/cli/run.go:365  retrying after error while creating load: failed to initialize the load generator: dial tcp 10.128.1.2:26257: connect: connection refused
		  | E200812 06:51:48.680821 1 workload/cli/run.go:379  Attempt to create load generator failed. It's been more than 10m0s since we started trying to create the load generator so we're giving up. Last failure: failed to initialize the load generator: dial tcp 10.128.1.2:26257: connect: connection refused
		  | Error: failed to initialize the load generator: dial tcp 10.128.1.2:26257: connect: connection refused
		  | Error: COMMAND_PROBLEM: exit status 1
		  | (1) COMMAND_PROBLEM
		  | Wraps: (2) Node 2. Command with error:
		  |   | ```
		  |   | ./workload run schemachange --verbose=1 --tolerate-errors=true --max-ops 100 --concurrency 5 {pgurl:1-4}
		  |   | ```
		  | Wraps: (3) exit status 1
		  | Error types: (1) errors.Cmd (2) *hintdetail.withDetail (3) *exec.ExitError
		  |
		  | stdout:
		Wraps: (5) exit status 20
		Error types: (1) *withstack.withStack (2) *safedetails.withSafeDetails (3) *errutil.withMessage (4) *main.withCommandDetails (5) *exec.ExitError

	cluster.go:1612,context.go:135,cluster.go:1601,test_runner.go:823: dead node detection: /home/agent/work/.go/src/github.com/cockroachdb/cockroach/bin/roachprod monitor teamcity-2172497-1597213671-09-n4cpu4 --oneshot --ignore-empty-nodes: exit status 1 4: 7426
		2: 8033
		3: dead
		1: 8046
		Error: UNCLASSIFIED_PROBLEM: 3: dead
		(1) UNCLASSIFIED_PROBLEM
		Wraps: (2) attached stack trace
		  | main.glob..func13
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1115
		  | main.wrap.func1
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:266
		  | github.com/spf13/cobra.(*Command).execute
		  | 	/home/agent/work/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:830
		  | github.com/spf13/cobra.(*Command).ExecuteC
		  | 	/home/agent/work/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:914
		  | github.com/spf13/cobra.(*Command).Execute
		  | 	/home/agent/work/.go/pkg/mod/github.com/spf13/cobra@v0.0.5/command.go:864
		  | main.main
		  | 	/home/agent/work/.go/src/github.com/cockroachdb/cockroach/pkg/cmd/roachprod/main.go:1808
		  | runtime.main
		  | 	/usr/local/go/src/runtime/proc.go:203
		  | runtime.goexit
		  | 	/usr/local/go/src/runtime/asm_amd64.s:1357
		Wraps: (3) 3 safe details enclosed
		Wraps: (4) 3: dead
		Error types: (1) errors.Unclassified (2) *withstack.withStack (3) *safedetails.withSafeDetails (4) *errors.errorString

```

<details><summary>More</summary><p>


Artifacts: [/schemachange/mixed-versions](https://teamcity.cockroachdb.com/viewLog.html?buildId=2172497&tab=artifacts#/schemachange/mixed-versions)
Related:
- #50928 roachtest: schemachange/mixed-versions failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-20.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-20.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #48406 roachtest: schemachange/mixed-versions failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-provisional_202005041945_v19.1.9](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-provisional_202005041945_v19.1.9) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #47980 roachtest: schemachange/mixed-versions failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.2](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.2) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

- #47959 roachtest: schemachange/mixed-versions failed  [C-test-failure](https://api.github.com/repos/cockroachdb/cockroach/labels/C-test-failure) [O-roachtest](https://api.github.com/repos/cockroachdb/cockroach/labels/O-roachtest) [O-robot](https://api.github.com/repos/cockroachdb/cockroach/labels/O-robot) [branch-release-19.1](https://api.github.com/repos/cockroachdb/cockroach/labels/branch-release-19.1) [release-blocker](https://api.github.com/repos/cockroachdb/cockroach/labels/release-blocker)

[See this test on roachdash](https://roachdash.crdb.dev/?filter=status%3Aopen+t%3A.%2Aschemachange%2Fmixed-versions.%2A&sort=title&restgroup=false&display=lastcommented+project)
<sub>powered by [pkg/cmd/internal/issues](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)</sub></p></details>

@otan this seems to be related to some geospatial types stuff. Seems to be missing an entry in this function:
https://github.com/cockroachdb/cockroach/blob/e049fda6bd2f9222788a78daf67f6ff492408db5/pkg/sql/types/types.go#L1402


```
panic: unexpected Family: Box2DFamily

goroutine 2994 [running]:
panic(0x4c14f40, 0xc000ac6c10)
	/usr/local/go/src/runtime/panic.go:722 +0x2c2 fp=0xc0033ef6e8 sp=0xc0033ef658 pc=0x7b0ae2
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).SQLStandardNameWithTypmod(0xc00308d740, 0x0, 0x0, 0x1, 0xc002b3fcd8)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1573 +0x1781 fp=0xc0033ef860 sp=0xc0033ef6e8 pc=0x1642761
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).SQLStandardName(0xc00308d740, 0xc002b3fcd8, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1390 +0x39 fp=0xc0033ef898 sp=0xc0033ef860 pc=0x1640f29
github.com/cockroachdb/cockroach/pkg/sql/types.(*T).InformationSchemaName(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/types/types.go:1587
github.com/cockroachdb/cockroach/pkg/sql.glob..func67.1.1(0xc002517000, 0x10, 0x48b8fa0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:411 +0x11d8 fp=0xc0033efb28 sp=0xc0033ef898 pc=0x3211bb8
github.com/cockroachdb/cockroach/pkg/sql.forEachColumnInTable(0xc0028f2500, 0xc0033efbb8, 0xc0038d5940, 0x78a88b)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1821 +0x83 fp=0xc0033efb60 sp=0xc0033efb28 pc=0x310f013
github.com/cockroachdb/cockroach/pkg/sql.glob..func67.1(0xc002654600, 0x500d55a, 0x6, 0xc0028f2500, 0x0, 0x0)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:378 +0x166 fp=0xc0033efc08 sp=0xc0033efb60 pc=0x3212036
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDesc.func1(0xc002654600, 0x500d55a, 0x6, 0xc0028f2500, 0xc0008959d0, 0x0, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1633 +0x4d fp=0xc0033efc48 sp=0xc0033efc08 pc=0x31ebe3d
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookupInternal(0x60e9860, 0xc0038d5940, 0xc00189a3f0, 0xc001a83320, 0x0, 0x4c9b100, 0xc0033efd88, 0xc000e9c3c0, 0x60)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1793 +0x421 fp=0xc0033efd40 sp=0xc0033efc48 pc=0x310ebe1
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDescWithTableLookup(...)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1698
github.com/cockroachdb/cockroach/pkg/sql.forEachTableDesc(0x60e9860, 0xc0038d5940, 0xc00189a3f0, 0xc001a83320, 0x0, 0xc000f625e8, 0x321b920, 0xc000f62610)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:1627 +0x8d fp=0xc0033efda8 sp=0xc0033efd40 pc=0x310e3cd
github.com/cockroachdb/cockroach/pkg/sql.glob..func67(0x60e9860, 0xc0038d5940, 0xc00189a3f0, 0xc001a83320, 0xc001a83380, 0xc000f62780, 0x7de670)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/information_schema.go:373 +0xa7 fp=0xc0033efe20 sp=0xc0033efda8 pc=0x31cac37
github.com/cockroachdb/cockroach/pkg/sql.virtualDefEntry.getPlanInfo.func1.1(0x6053720, 0xc00378af20, 0x2, 0x1)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_schema.go:484 +0x11a fp=0xc0033efe98 sp=0xc0033efe20 pc=0x3201f0a
github.com/cockroachdb/cockroach/pkg/sql.setupGenerator.func2(0x60e9860, 0xc0038d5ac0, 0xc0003399e0, 0xc003665b00, 0xc00378af20)
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:112 +0x109 fp=0xc0033effb8 sp=0xc0033efe98 pc=0x3203b69
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:1357 +0x1 fp=0xc0033effc0 sp=0xc0033effb8 pc=0x7e12b1
created by github.com/cockroachdb/cockroach/pkg/sql.setupGenerator
	/go/src/github.com/cockroachdb/cockroach/pkg/sql/virtual_table.go:101 +0xf7
```

--- a/pkg/sql/types/types.go
+++ b/pkg/sql/types/types.go
@@ -1436,6 +1436,8 @@ func (t *T) SQLStandardNameWithTypmod(haveTypmod bool, typmod int) string {
 		return buf.String()
 	case BoolFamily:
 		return "boolean"
+	case Box2DFamily:
+		return "box2d"
 	case BytesFamily:
 		return "bytea"
 	case DateFamily:

--- a/pkg/sql/types/types_test.go
+++ b/pkg/sql/types/types_test.go
@@ -21,6 +21,7 @@ import (
 	"github.com/cockroachdb/cockroach/pkg/util/protoutil"
 	"github.com/lib/pq/oid"
 	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
 )
 
 func TestTypes(t *testing.T) {
@@ -971,6 +972,14 @@ func TestOidSetDuringUpgrade(t *testing.T) {
 			err := input.upgradeType()
 			assert.NoError(t, err)
 			assert.Equal(t, Oid, input.Oid())
+		})
+	}
+}
+
+func TestSQLStandardName(t *testing.T) {
+	for _, typ := range Scalar {
+		t.Run(typ.Name(), func(t *testing.T) {
+			require.NotEmpty(t, typ.SQLStandardName())
 		})
 	}
 }

